src/Integration.UnitTests/SonarAnalyzer/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -0,0 +1,101 @@
+        {
+            this.serviceProvider = new ConfigurableServiceProvider(false);
+            this.host = new ConfigurableHost(this.serviceProvider, Dispatcher.CurrentDispatcher);
+            var mefExport1 = MefTestHelpers.CreateExport<IHost>(this.host);
+
+            this.activeSolutionTracker = new ConfigurableActiveSolutionTracker();
+            var mefExport2 = MefTestHelpers.CreateExport<IActiveSolutionTracker>(this.activeSolutionTracker);
+
+            var mefModel = ConfigurableComponentModel.CreateWithExports(mefExport1, mefExport2);
+
+            this.serviceProvider.RegisterService(typeof(SComponentModel), mefModel, replaceExisting: true);
+
+            this.solutionMock = new SolutionMock();
+            this.serviceProvider.RegisterService(typeof(SVsSolution), this.solutionMock);
+        }
+
+        [TestMethod]
+        public void ActiveSolutionBoundTracker_Unbound()
+        {
+            // Setup
+            var solutionBinding = new ConfigurableSolutionBinding()
+            {
+                CurrentBinding = null
+            };
+            this.serviceProvider.RegisterService(typeof(ISolutionBinding), solutionBinding);
+            host.VisualStateManager.SetBoundProject(new ProjectInformation());
+            var testSubject = new ActiveSolutionBoundTracker(this.serviceProvider);
+
+            // Verify
+            Assert.IsFalse(testSubject.IsActiveSolutionBound, "Unbound solution should report false activation");
+        }
+
+        [TestMethod]
+        public void ActiveSolutionBoundTracker_Bound()
+        {
+            // Setup
+            var solutionBinding = new ConfigurableSolutionBinding()
+            {
+                CurrentBinding = new BoundSonarQubeProject()
+            };
+            this.serviceProvider.RegisterService(typeof(ISolutionBinding), solutionBinding);
+            host.VisualStateManager.SetBoundProject(new ProjectInformation());
+            var testSubject = new ActiveSolutionBoundTracker(this.serviceProvider);
+
+            // Verify
+            Assert.IsTrue(testSubject.IsActiveSolutionBound, "Bound solution should report true activation");
+        }
+
+        [TestMethod]
+        public void ActiveSolutionBoundTracker_Changes()
+        {
+            // Setup
+            var solutionBinding = new ConfigurableSolutionBinding()
+            {
+                CurrentBinding = new BoundSonarQubeProject()
+            };
+            this.serviceProvider.RegisterService(typeof(ISolutionBinding), solutionBinding);
+            host.VisualStateManager.SetBoundProject(new ProjectInformation());
+
+            var testSubject = new ActiveSolutionBoundTracker(this.serviceProvider);
+
+            // Act & Verify
+            Assert.IsTrue(testSubject.IsActiveSolutionBound, "Bound solution should report true activation");
+            // todo: this is not nice that we need to call two things, why isn't the ClearBoundProject() clear the CurrentBinding
```

Comment:
The answer is simple - those are mocks of the project and are separate components that are integrated together, which you don't have in tests. So I don't think it is a problem.